### ISSUE TITLE
Prebid JS vs. DFP Reporting

### ISSUE THREAD

**Issue by testerman619** [prebid/Prebid.js#887]
Hi. Pardon my lack of knowledge in this area. We're just getting started with Prebid JS for header bidding.

We fallback to AdX for a last look and AdX wins half the time.

I had a few questions regarding reporting in DFP relative to the PBJS messages that are sent. We are able to  tie out the cpm from the bid won event to the Ad Server Revenue in DFP without issue. We use use HB_Bidder=**** line items to track revenue by bidders. My question involves the total code served and possibly AdX revenue. In theory, the total code served for HB_Bidder should be equal to the distinct count of requestId+ placementCode from bid requested, but the distinct count is 40% higher than DFP.

Furthermore, i'm trying to understand the AdX impressions and revenue. I would assume that the total code served minus bids won from pbjs.getBidResponses()  would be the AdX impressions, but it's way off.

Ultimately we would like to get an educated realtime guess of AdX revenue (assuming they fill the impressions slightly over the highest bid won). Has anyone had luck with this?

**Comment by mkendall07** [comment]
I'm not clear on your terminology, can you clarify what you mean by "code served"? The way I understand it, you have 3 primary buckets, Prebid won, AdX won or Direct booked won. Out of these the impressions should be distributed. We don't have a way to report on anything in DFP as this is a client side library.

**Comment by testerman619** [comment]
Sorry. I have a very basic knowledge of prebid at this point.

Total Code Served is the DFP term for the impressions that were bid on (or not filled). They report Ad Server impressions (prebid won) impressions and AdX impressions. The Prebid Won ties directly to the ad server impressions. We do not have Direct Booked, so not a concern.

What we're trying to determine is the AdX won from the prebid messaging, or possibly a pass back from DFP. We see a much smaller total code served than requests (there may be some issues with our reporting), but we're trying to hone in on AdX wins.

Hope that help, thanks!

**Comment by chuckeatworld24** [comment]
Happy to help but very confused on your question still! Can you rephrase/be more concise? If you want ADX reporting, this is available via your ADX Account (which can also be accessed via your DFP Account per recent DFP Updates)

The metrics you are referencing are inherently flawed as far as accuracy and you should use the newer metric "Ad Server Downloaded Impressions" --- the others count impressions based on less accurate standards (they look at the <head> code and not the <body>/<div> tag of the DFP Tag)
